build.mill

```diff
@@ -0,0 +1,435 @@
+    ).call()
+    os.read.lines(f).filter(s => s.nonEmpty && !s.startsWith("#"))
+  }
+
+  override def generatedSources: T[Seq[PathRef]] = Task {
+    super.generatedSources() ++ {
+      // @formatter:off
+      os.proc(
+        Seq(jextractBinary().toString, header().path.toString)
+          ++ includePaths().flatMap(p => Seq("-I", p.path.toString))
+          ++ Seq(
+            "-t", target(),
+            "--header-class-name", headerClassName(),
+            "--source",
+            "--output", Task.dest.toString
+          ) ++ includeFunctions().flatMap(f => Seq("--include-function", f)) ++
+          includeConstants().flatMap(f => Seq("--include-constant", f)) ++
+          includeStructs().flatMap(f => Seq("--include-struct", f)) ++
+          includeTypedefs().flatMap(f => Seq("--include-typedef", f)) ++
+          includeUnions().flatMap(f => Seq("--include-union", f)) ++
+          includeVars().flatMap(f => Seq("--include-var", f)) ++
+          linkLibraries().flatMap(l => Seq("-l", l))
+      ).call(Task.dest)
+      // @formatter:on
+      Seq(PathRef(Task.dest))
+    }
+  }
+
+  override def javacOptions = Task(super.javacOptions() ++ Seq("--enable-preview", "--release", "21"))
+}
+
+trait HasCIRCTPanamaBindingModule extends JavaModule {
+  import $file.circtpanamabinding.CIRCTPanamaBinding
+  def circtPanamaBindingModule: CIRCTPanamaBinding
+
+  override def moduleDeps = super.moduleDeps ++ Some(circtPanamaBindingModule)
+  //
+  override def javacOptions = Task(super.javacOptions() ++ Seq("--enable-preview", "--release", "21"))
+
+  override def forkArgs: T[Seq[String]] = Task(
+    super.forkArgs() ++ Seq("--enable-native-access=ALL-UNNAMED", "--enable-preview")
+      ++ circtPanamaBindingModule
+        .libraryPaths()
+        .map(p => s"-Djava.library.path=${p.path}")
+  )
+}
+
+trait HasPanamaLibModule extends ScalaModule with HasCIRCTPanamaBindingModule {
+  import $file.panamalib.PanamaLib
+  def panamaLibModule: PanamaLib
+
+  def circtPanamaBindingModule = panamaLibModule.circtPanamaBindingModule
+
+  override def moduleDeps = super.moduleDeps ++ Some(panamaLibModule)
+}
+
+trait HasPanamaOMModule extends ScalaModule with HasCIRCTPanamaBindingModule {
+  import $file.panamaom.PanamaOM
+  def panamaOMModule: PanamaOM
+
+  def circtPanamaBindingModule = panamaOMModule.circtPanamaBindingModule
+
+  override def moduleDeps = super.moduleDeps ++ Some(panamaOMModule)
+}
+
+trait HasPanamaConverterModule extends ScalaModule with HasCIRCTPanamaBindingModule {
+  import $file.panamaconverter.PanamaConverter
+  def panamaConverterModule: PanamaConverter
+
+  def circtPanamaBindingModule = panamaConverterModule.circtPanamaBindingModule
+
+  override def moduleDeps = super.moduleDeps ++ Some(panamaConverterModule)
+}
+
+object chisel extends Cross[Chisel](v.scalaCrossVersions)
+
+trait Chisel extends CrossSbtModule with HasScala2MacroAnno with HasScala2Plugin with ScalafmtModule {
```

Comment:
Maybe we should have another package.mill in the root folder for `chisel` source?

Reply:
We cannot do this at the moment because `package.mill` files are only discovered in subfolders, but once we drop the sbt flow we can move chisel sources to the subfolder like other modules.